les/access/context.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2015 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+// Package access provides a layer to handle local blockchain database and
+// on-demand network retrieval
+package access
+
+import (
+	"sync/atomic"
+	"time"
+)
+
+// Context for ODR requests, can be timed out or cancelled manually
+type OdrContext struct {
+	cancel, cancelOrTimeout chan struct{}
+	id                      *OdrChannelID
+	cancelled               int32
+}
+
+var NullCtx = (*OdrContext)(nil) // used when creating states
+var NoOdr = (*OdrContext)(nil)   // used for individual requests
```

Comment:
I don't like wrapping `nil`s in various names. It makes the code much harder to read since you need to keep thinking (and look it up for the first time) what these mean. If it's a nil pointer, then simply use `nil` throughout the code and don't give it a fancy name.